C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
+  # string: '(f|s|raw)?\"[^\"]*\"'
+  # literal: '({{number}}|{{char}}|{{string}})'
+
+  visibility: \b(public|private|protected|internal|protected\s+internal)\b
+  base_type: (?:bool|byte|sbyte|char|decimal|double|float|int|uint|long|ulong|short|ushort|object|string|void)
+  type_suffix: ((?:\[,*\]|\*|\?)*)
+  generic_declaration: \s*(<[^(={};<]*>)?\s*
+  reserved: '(?:abstract|as|base|break|case|catch|checked|class|const|continue|default|delegate|do|else|enum|event|explicit|extern|finally|fixed|for|foreach|goto|if|implicit|in|interface|internal|is|lock|namespace|new|null|operator|out|override|params|private|protected|public|readonly|ref|return|sealed|sizeof|stackalloc|static|string|struct|switch|this|throw|try|typeof|unchecked|unsafe|using|virtual|volatile|while)'
+  name: '(@{{reserved}}|@{{base_type}}|@var|{{name_normal}})'
+  # name: '([A-Za-z_][A-Za-z_0-9]*)'
+  # name_normal: '[A-Za-z_][A-Za-z_0-9]*'
+  # name_normal: '(?:[A-Za-z_\p{L}]|unicode_char)[A-Za-z_0-9\p{L}]*'
+  start_char: '(?:{{unicode_char}}|[A-Za-z_\p{L}])'
+  other_char: '(?:{{unicode_char}}|[A-Za-z_0-9\p{L}])'
+  name_normal: '{{start_char}}{{other_char}}*'
+  cap_name: '([A-Z]{{other_char}})'
+
 contexts:
-  main:
-    - match: '^\s*(using)\s+([^ ;]*);'
-      scope: meta.keyword.using.source.cs
-      captures:
-        1: keyword.other.using.source.cs
-    - match: '^\s*((namespace)\s+([\w.]+))'
+  prototype:
+    - include: comments
+    - include: git_merge
+    - match: '#\s*'
+      push: preprocessor_option
+
+    # comments
+  comments:
+    - match: '^\s*(///)'
       captures:
-        1: meta.namespace.identifier.source.cs
-        2: keyword.other.namespace.source.cs
-        3: entity.name.type.namespace.source.cs
+        1: punctuation.definition.comment.documentation.cs
+      push: documentation
+    - match: '//'
+      scope: punctuation.definition.comment.cs
       push:
-        - meta_scope: meta.namespace.source.cs
-        - match: "}"
-          scope: punctuation.section.namespace.end.source.cs
+        - meta_scope: comment.line.double-slash.cs
+        - match: $
           pop: true
-        - match: "{"
-          scope: punctuation.section.namespace.begin.source.cs
-          push:
-            - meta_scope: meta.namespace.body.source.cs
-            - match: "(?=})"
-              pop: true
-            - include: code
-    - include: code
-  block:
-    - match: "{"
-      scope: punctuation.section.block.begin.source.cs
+        - include: comments_in
+    - match: '/\*'
+      scope: punctuation.definition.comment.begin.cs
+      push:
+        - meta_scope: comment.block.cs
+        - match: '\*/'
+          scope: punctuation.definition.comment.end.cs
+          pop: true
+        - include: comments_in
+
+  comments_in:
+    - match: \b(?i:todo|hack|xxx)\b
+      scope: comment.line.todo
+    - match: '/!\\'
+      scope: comment.line.todo
+    - match: \b(https?://\S*)\b
+      scope: markup.underline.link.cs
+
+  git_merge:
```

Comment:
I don't want to try to get into highlighting things like this in the default syntaxes.

Ideally this can be handled by a plugin that will scan the file and look for such patterns and alert the user.